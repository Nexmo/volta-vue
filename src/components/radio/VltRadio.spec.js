import { shallowMount } from '@vue/test-utils';
import { VltRadio } from '..';

describe('vlt-radio', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(VltRadio, {
      propsData: {
        label: 'Do I pass?',
        name: 'pass',
        val: 'Yes',
      },
    });
  });

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders correctly with label "Do I pass?", name "pass" and value "Yes"', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('emits "yes" on value change', () => {
    wrapper.find('input').trigger('click');
    expect(wrapper.emitted('input')).toEqual([['Yes']]);
  });
});
