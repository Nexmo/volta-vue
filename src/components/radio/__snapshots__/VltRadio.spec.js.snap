// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-radio renders correctly with label "Do I pass?", name "pass" and value "Yes" 1`] = `
<div
  class="Vlt-radio"
>
  <label>
    <span
      class="Vlt-radio__button"
    >
      <input
        name="pass"
        type="radio"
        value="Yes"
      />
       
      <span
        class="Vlt-radio__icon"
      />
    </span>
     
    <span>
      
      Do I pass?
      
      <!---->
    </span>
  </label>
</div>
`;
